### PR TITLE
feature/CasysAndPreprocessingAndXAlign

### PR DESCRIPTION
**Pull request by Sandy-CHERY**
## Description

**Fixes 1 Change Cassys to solve the max stack size reached problem**

## Motivation and Context

Alexis Neme have a problem :
"When I execute Cassys  command (below) with IDE it generates the error "Max stack size reached" (attcahed file)
"Local\Unitex\App\UnitexToolLogger.exe" Cassys "-aUnitex3_2\English\Alphabet.txt" "-tUnitex3_2\English\Corpus\text.snt" "-lUnitex3_2\English\CasSys\cascade.csc" -v "-rUnitex3_2\English\Graphs\" "--input_offsets=Unitex3_2\English\Corpus\text_snt\normalize.out.offsets" -qutf8-no-bom

When I execute Cassys  command (below) with a command line, it does generate an error
"Local\Unitex\App\UnitexToolLogger.exe" Cassys "-aUnitex3_2\English\Alphabet.txt" "-tUnitex3_2\English\Corpus\text.snt" "-lUnitex3_2\English\CasSys\cascade.csc" -v "-rUnitex3_2\English\Graphs\" "--input_offsets=Unitex3_2\English\Corpus\text_snt\normalize.out.offsets" -qutf8-no-bom --locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000

I propose to include these parameters in bold to the Cassys command line call from the IDE"

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\process\commands\CommandBuilder.java,
when we create the Cassys comand "-qutf8-no-bom", I add "-qutf8-no-bom --locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000"

## Description

**Fixes 2 Preprocessing**

## Motivation and Context

#105 opened on 28 May 2018 by denisMaurel : SET (line Merge) in Preprocessing window type:feature-request

When you lanch the Preprocessing, graph in MERGE mode and graph in REPLACE mode have the same path (Graphs\Preprocessing\Replace)

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\frames\PreprocessDialog.java
Sentence and Replace used the same variable, I creat two different variables for having two different paths (Graphs\Preprocessing\Sentence and Graphs\Preprocessing\Replace).

## Description

**Fixes 3 Change XAlign problem**

## Motivation and Context

When you lanch XAlign, you need to choose two text, Source and Target. If texts are in .txt, you need to choose the .XML text. but they have the same path.

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\frames\XAlignConfigFrame.java
Source and Target used the same variable saveXMLChooser, I creat two different variables for having two different paths.

### CODE REVIEW

File: XAlignConfigFrame.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
+/*
+ * Unitex
+ *
+ * Copyright (C) 2001-2019 Université Paris-Est Marne-la-Vallée <unitex@univ-mlv.fr>
```
Comment (martinec) [inline XAlignConfigFrame.java:4]: 2001-**2021**

File: XAlignConfigFrame.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+			JOptionPane.showMessageDialog(null, "Source text not found!",
+					"Error", JOptionPane.ERROR_MESSAGE);
+			return;
+		}
+		if (source.getAbsolutePath().endsWith("txt")) {
+			/* If the user wants to open a raw text file */
+			JOptionPane
+					.showMessageDialog(
+							null,
+							"Your source file is a .txt one. Please select the\n"
```
Comment (martinec) [inline XAlignConfigFrame.java:166]: What about:

```
Your source file is in .txt format. 
Please select the destination file that will be used by XAlign (TEI format).
```

File: PreprocessDialog.java [hunk 3]
```diff
@@ -0,0 +1,705 @@
+/*
```
Comment (martinec) [inline PreprocessDialog.java:1]: It seems that this file is already under `unitex/src/en/umlv/unitex/frames/PreprocessDialog.java` is there any reason to try to create a new copy under the root?

File: XAlignConfigFrame.java [hunk 4]
```diff
@@ -0,0 +1,341 @@
+/*
```
Comment (martinec) [inline XAlignConfigFrame.java:1]: It seems that this file is already under `unitex/src/fr/umlv/unitex/frames/XAlignConfigFrame.java` is there any reason to try to create a new copy under the root?

File: config/Config.java [hunk 5]
```diff
@@ -0,0 +1,1646 @@
+/*
```
Comment (martinec) [inline config/Config.java:1]: It seems that this file is already under `unitex/src/fr/umlv/unitex/config/Config.java`  is there any reason to try to create a new copy under `config/Config.java`?

File: process/commands/AbstractCommand.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (martinec) [inline process/commands/AbstractCommand.java:1]: It seems that this file is already under `/unitex/src/fr/umlv/unitex/process/commands/AbstractCommand.java` is there any reason to try to create a new copy under `process/commands/AbstractCommand.java`?

File: process/commands/CommandBuilder.java [hunk 7]
```diff
@@ -0,0 +1,319 @@
+/*
```
Comment (martinec) [inline process/commands/CommandBuilder.java:1]: It seems that this file is already under `unitex/src/fr/umlv/unitex/process/commands/CommandBuilder.java` is there any reason to try to create a new copy under `process/commands/CommandBuilder.java`?

**Review by martinec — CHANGES_REQUESTED** [review]
It seems that the files in your PR have been modified and then placed in the wrong place.

At this point, it is not possible to compile the code, nor to visualize the changes made.

Please update your PR or send a new one with the necessary adjustments.

**Comment by Sandy-CHERY** [comment]
Bonjour,
Peux-tu m'aider, je ne parviens pas a mettre a jour la PR.
Est-ce que je dois la fermer et en refaire une ?

**Comment by martinec** [comment]
> It seems that the files in your PR have been modified and then placed in the wrong place.
> 
> At this point, it is not possible to compile the code, nor to visualize the changes made.
> 
> Please update your PR or send a new one with the necessary adjustments.

As stated before, for this specific PR, if you wish, you can close it and create a new one.

**Pull request closed without merging**